File: python/day-1/part_1.py
with open("./python/day-1/input.txt") as file:
    lines = [line.strip() for line in file.readlines()]

position = 50
count = 0 

for line in lines:
    direction = line[0]
    turns = int(line[1:])

    if direction == "L":
        position -= turns
    else:
        position += turns

    position = position%100

    if position == 0:
        count += 1


print(count)

File: c++/day-1/part_1.cpp
#include <fstream>
#include <iostream>
#include <string>

int main()
{
    std::ifstream file("C:/Users/User/advent-od-code/c++/day-1/input.txt");

    std::string line;
    int turns = 0;
    enum Direction
    {
        Right = 'R',
        Left = 'L'
    };
    int count = 0;
    int position = 50;

    if (file.is_open())
    {
        while (getline(file, line))
        {
            Direction direction = static_cast<Direction>(line[0]);
            turns = std::stoi(line.substr(1));
            if (direction == Direction::Left)
            {
                position -= turns;
            }
            else
            {
                position += turns;
            }
            position = position % 100;
            if (position == 0)
            {
                count += 1;
            }

        }
    }

    file.close();

    std::cout << count << std::endl;
}